Client/create/src/Pin.js

```diff
@@ -33,6 +35,10 @@ module.exports = React.createClass({displayName: "exports",
       this.setState({editLocationMode: true});
     }
   },
+  removePin:function(){
+    Actions.removePin(this.props.pinIndex);
+    this.map.remove(this.props.pinIndex);
```

Comment:
I am not sure if this will actually get called as Actions triggers an immediate reflow. Have you verified this? You can try and console log after line 39 to check.